jax/_src/numpy/lax_numpy.py

```diff
@@ -2543,17 +2543,41 @@ def _pad(array, pad_width, mode, constant_values, stat_length, end_values, refle
     return _pad_empty(array, pad_width)
 
   else:
-    msg = "Unimplemented padding mode '{}' for np.pad."
```

Comment:
Shouldn't this be an "Unimplemented padding mode" error? I believe that the previous version of your changes had the right `else` clause here, followed by the `assert False`

Reply:
I am not mistaken,
https://github.com/google/jax/pull/5229/files#diff-13c884892e20a9f28e3bd980ee444f3b0d5d90258fe9aa6dcfa84a2089d34844R2570
I moved "Unimplemented padding mode" to above.

Every errors of unsupported modes and invalid keyword arguments are handled in `pad` function,
I thought `_pad` function should never reach its end.

Reply:
Indeed, sorry!

Reply:
Right